### ISSUE TITLE
Fix PyTorch weights_only warning and PyAnnote checkpoint upgrade warnings in stitch step

### ISSUE THREAD

**Issue by aengus-signal4** [aengus-signal4/signal4-core#16]
## Problem

The stitch processing step is generating multiple warnings that are being recorded as failures by the failure tracker:

### 1. PyTorch `weights_only=False` FutureWarning

```
/Users/signal4/signal4/core/src/processing_steps/stitch.py:49: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling...
```

In a future PyTorch release, `weights_only` will default to `True`. We should explicitly set this parameter to avoid breaking changes.

### 2. PyAnnote/PyTorch Lightning Checkpoint Warnings

```
Lightning automatically upgraded your loaded checkpoint from v1.2.7 to v2.6.0. To apply the upgrade to your files permanently, run `python -m pytorch_lightning.utilities.upgrade_checkpoint ../../.cache/torch/pyannote/models--pyannote--embedding/snapshots/4db4899737a38b2d618bbd74350915aa10293cb2/pytorch_model.bin`
```

Multiple warnings related to:
- `ModelCheckpoint` callback state key collisions
- Task-dependent loss function warning (suggests setting `strict=False`)
- Missing keys in model state dict: `['loss_func.W']`

## Impact

These warnings are being captured by `failure_tracker` and recorded as failures, even though the task may complete successfully. This could lead to:
- Unnecessary task retries
- Confusing failure statistics
- Potential future breakage when PyTorch changes defaults

## Suggested Fixes

1. **For `torch.load` warning**: Explicitly pass `weights_only=True` (or `False` if unpickling custom objects is required) in `stitch.py:49`

2. **For PyAnnote checkpoint warnings**: 
   - Run the suggested upgrade command on all workers: `python -m pytorch_lightning.utilities.upgrade_checkpoint ~/.cache/torch/pyannote/models--pyannote--embedding/snapshots/4db4899737a38b2d618bbd74350915aa10293cb2/pytorch_model.bin`
   - Or suppress these specific warnings if they don't affect functionality

3. **Consider filtering warnings**: Update failure tracker to not record warnings-as-errors for known benign warnings

## Location

- `src/processing_steps/stitch.py:49`
- Affects: stitch pipeline step on all workers
